core/src/main/java/hudson/PluginManager.java

```diff
@@ -1643,26 +1649,73 @@ public HttpResponse doUploadPlugin(StaplerRequest req) throws IOException, Servl
     @Restricted(NoExternalUse.class)
     @RequirePOST public HttpResponse doCheckUpdatesServer() throws IOException {
         Jenkins.getInstance().checkPermission(Jenkins.ADMINISTER);
-        try {
-            for (UpdateSite site : Jenkins.getInstance().getUpdateCenter().getSites()) {
-                FormValidation v = site.updateDirectlyNow(DownloadService.signatureCheck);
-                if (v.kind != FormValidation.Kind.OK) {
-                    // TODO crude but enough for now
-                    return v;
-                }
+
+        int sleep = SystemProperties.getInteger(PluginManager.class.getName() + ".sleepTimeMilis", 1000);
+        int retries = SystemProperties.getInteger(PluginManager.class.getName() + ".retries", 3);
```

Comment:
It might be clearer if it indicated what operation these variables relate to. Maybe something like "checkUpdatesSleepTimeMillis" and "checkUpdatesRetries".

Reply:
+1

Reply:
@MRamonLeon, @batmat is right. In general this should be read in a static block when the class is initialized